rfcs/0027-trusted-bots.md

```diff
@@ -0,0 +1,75 @@
+---
+feature: trusted-bots
+start-date: 2018-03-23
+author: Matthew Bauer
+co-authors:
+related-issues: https://github.com/matthewbauer/rfcs/pull/27
+---
+
+# Summary
+[summary]: #summary
+
+Create a "trusted bot" process to manage GitHub bots. Bot maintainers can apply to make their bot "trusted", adding it to the bot team.
+
+# Motivation
+[motivation]: #motivation
+
+Why do we need bots? Nix has never had the accessibility of a package manager like Homebrew or the massive user base of a distro like Debian. To compete with them, we need to make our developers more productive. Bots can be a massive help.
+
+Currently running bots:
+
+- @GrahamC's @GrahamCOfBorg
+- @ryantm's [nix-update](https://github.com/ryantm/nix-update)
+- Flying Circus's [Vulnix](https://github.com/flyingcircusio/vulnix)
+
+Past bots:
+
+- Travis CI bot
+- [mention-bot](https://github.com/facebook/mention-bot)
+
+Right now we have kind of a wild west of bots where anyone can create and manage their own bots. This is great for iterating on new ideas, but has certain dangers. The biggest danger is that these maintainers get hit by a bus and are unable to continue managing the bot. In the long-term, my goal is to have these bots managed by the NixOS organization. The trusted bots program is the first step in that direction.
```

Comment:
Re: wild west: I think this sets a slightly wrong tone. Is it wild west that we use different editors? With some private keybindings to accelearte common Nix editor tasks? With some wrapper scripts to help do routine nix-related tasks?

Bots requiring special access (more than an external user) are a separate topic, but the only bot that has special non-human-gated access right now has got a separate account and member status, so not exactly wild west.

Maybe the framing should be «deployment best practices and a process for recognition the bots, directly or indirectly used by many Nixpkgs developers, that have become a valuable community resource»? Maybe with «attracting maintenance help» angle of putting the repositories under the organisation umbrella, maybe with a process of infrastructure management by NixOS Foundation for the most critical tools (or tools needing to run autonomously) — that part needs feedback from people managing Foundation, of course.

Re: bus: let's be honest and notice that the largest risk for a bot — and also the largest risk for a package! — is that the author becomes too busy with other things and has no time to maintain it.